### ISSUE TITLE
Polynomial Regression Plot Renders as Zigzagged Line

### ISSUE THREAD

**Issue by aviralvardiya** [microsoft/ML-For-Beginners#968]
### Description

In the Polynomial Regression section of `2-Regression/3-Linear/README.md`, the prediction curve appears smooth in the included graph image. However, if learners run the code `plt.plot(X_test, pred)` (as taught previously in the lesson) to replicate the polynomial regression prediction plot, they get a zigzagged, scribbled line instead of a smooth curve.

<img width="548" height="413" alt="Image" src="https://github.com/user-attachments/assets/a2a2262b-2f84-4791-8388-b5cc8ad8699d" />

In isolation the curve looks like:

<img width="561" height="413" alt="Image" src="https://github.com/user-attachments/assets/105c1a07-4c49-4d10-b5f1-b313b9569676" />

This happens because:
- `train_test_split()` shuffles the data randomly before splitting into train/test sets
- The values in `X_test` are therefore not in sequential order
- When `plt.plot()` connects points in the order they appear in the array, it creates a zigzag pattern rather than a smooth curve

### Expected Behavior

The plot should display a smooth polynomial curve that correctly visualizes the model's prediction across the data range.

### Solution

Instead of plotting directly on unsorted test data, create a uniform range of input values using `np.linspace`:

```python
X_range = np.linspace(X_test.min(), X_test.max(), 100).reshape(-1,1)
y_range = pipeline.predict(X_range)

plt.scatter(X_test, y_test)
plt.plot(X_range, y_range)
```

This creates 100 evenly-spaced points across the test data range, producing a smooth curve that properly represents the polynomial regression model.

**Comment by aviralvardiya** [comment]
Please let me know if this issue is worthwhile. I'll create a PR for the same.

**Comment by maryamzamani02-lang** [comment]
> Please let me know if this issue is worthwhile. I'll create a PR for the same.

**Comment by leestott** [comment]
Thanks for the report
